### DISCUSSION TITLE
how about support groubbatchsampler??

### DISCUSSION THREAD

**Discussion by beomgonyu** [Lightning-AI/pytorch-lightning#13083] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
When the size of the data changes significantly, its better to sorting, grouping, then batch within a group.
then batchnorm will works more stable, and training speed is better, I  think.

However, it seemed that pl didnt support grouping in sampler.

below is the regarding code.

```
train_sampler = torch.utils.data.RandomSampler(self.train_dataset)

group_ids = create_area_groups(self.train_dataset, k=self.groups)   
self.train_batch_sampler = GroupedBatchSampler(train_sampler, group_ids, 
                                               self.batch_size, drop_last=self.drop_last)

def train_dataloader(self) :
        return torch.utils.data.DataLoader(
            dataset=self.train_dataset,
            batch_sampler=self.train_batch_sampler,
            num_workers=self.workers,
            collate_fn=collate_fn,
            pin_memory=True,
        )
```

but I've got error in _dataloader_init_kwargs_resolve_sampler function(utilities.data.py )
I just  add argument  group_ids,  and training is going on well.

```
        batch_sampler = type(batch_sampler)(
            sampler,
            batch_size=batch_sampler.batch_size,
            group_ids=batch_sampler.group_ids,   <-- add this part
            drop_last=(False if is_predicting else batch_sampler.drop_last),
        )
```

How about support this in lightning officially.
thanks.
